src/Base/Exception.h

```diff
@@ -33,6 +33,13 @@
 #include "FileInfo.h"
 #include "BaseClass.h"
 
+/* MACROS FOR THROWING EXCEPTIONS */
+
+#define THROW(exception) throw Base::exception(__FILE__,__LINE__,__PRETTY_FUNCTION__);
+#define THROWM(exception, message) throw Base::exception(message,__FILE__,__LINE__,__PRETTY_FUNCTION__);

```

Comment:
@wwmayer 
I think both of your remarks make a lot of sense. I will review the code with those in mind.

@DeepSOIC 
Are you interested in using these macros for your exceptions at this time? If yes, should I move them out of Exception.h? If yes, where to?

@ianrrees 

I am not translating the message on exception throwing. Stripping strings is something I am trying to avoid. I have been there. It is not a nice solution. I go with Werner's proposals.

There are two reasons for wanting to add this "debug information":

1. Assisting in the handling of bug reports (mainly related to release builds), not only for those who code, but also for those who assist and manage the bug tracker. Many tickets go changing workbench as it is difficult for the reporter and those who manage and assist with the bug tracker, who (generally) do not code. Many reporters can not produce a trace because of the lack of debugging symbols and/or competence. In some cases where it is not easy to find the sequence of actions/configuration to reproduce the bug, having this information would help.

2. Assist the developers in the debugging of complex functionalities. Do a cursory search of the FC code and see how many exceptions include the name of the function in the exception text ATM. Many of these functions throw several exceptions. Sometimes with the same text. I can agree that this is mostly useful for debug builds. However, when something is complex to test, bugs are ultimately discovered by the community... using release builds.

There is always a trade off, I think it is useful to have that information and I am aware that there is some cost (I do not really expect this to increase the executable space much. It is maybe an increase of 70 characters per exception. How much is that going to be in the whole FC?). In any case, if we go ahead with it and we end up with the impression that it is not useful in release mode, then we can make some information only be compiled for Debug builds.

About the amount of code to maintain, with Werner's proposals there won't be much code to maintain.

Now, this is about improving exceptions. If we are not improving them, then better invest the time elsewhere. I see your reticence, so I am listening to your arguments carefully so as to arrive to better solution.